pipeline/price_modelling/modeler_bagged_elasticnet.py

```diff
@@ -0,0 +1,155 @@
+    return harmonised_data
+
+
+def get_topic_variables():
+    ''' Function to get all the names of the topic model.
+    '''
+
+    topic_variables = (
+        pd.read_sql('PRAGMA table_info(TopicModel)', engine)['name']
+        .where(lambda x: x != 'id')
+        .dropna()
+        .tolist())
+    cmoplete_topic_variable = [v + suffix
+                               for v in topic_variables
+                               for suffix in ['_pos', '_neg']]
+    return cmoplete_topic_variable
+
+
+def transform_harmonised_data(data, forecast_period, topic_variables,
+                              response_variable, filter_coef):
+    '''Function to transform the harmonised data for modeling.
+
+    The following transformation are performed:
+
+    1. Remove commodity tag (this is a temporary hack)
+    2. Create the response by shifting the price time serie.
+    3. Perform filtering.
+
+    '''
+    non_commodity_tag_columns = [i
+                                 for i in data.columns
+                                 if 'contain' not in i]
+    transformed_data = data.copy()[non_commodity_tag_columns]
+    transformed_data[response_variable] = (
+        transformed_data['GOI'].shift(-forecast_period))
+
+    # Perform cumsum, the implementation in R is more flexible where a
+    # filter cefficient can be applied.
+
+    for v in topic_variables:
+        transformed_data[v] = transformed_data[v].cumsum()
+
+    return transformed_data
+
+
+def build_datasets(complete_data, forecast_period, holdout_period,
+                   response_variable):
+    ''' Function to split the complete data into train, test, and prediction data.
+    '''
+
+    prediction_data = complete_data[complete_data[response_variable].isnull()]
+    model_data = complete_data[~complete_data[response_variable].isnull()]
+    train_data = model_data.iloc[1:-(holdout_period - 1)]
+    test_data = model_data.iloc[-(holdout_period):]
+    cutoff_date = train_data['date'].max() + timedelta(days=forecast_period)
+    return prediction_data, train_data, test_data, cutoff_date
+
+
+def train_bag_elasticnet(complete_data, forecast_period, holdout_period,
+                         bootstrapIteration, topic_variables,
+                         response_variable, date_col='date'):
+    '''Function to bag and train the Elastic net iwth cross-validation
+    error as weight.
+
+    '''
+
+    prediction_data, train_data, test_data, cutoff_date = (
+        build_datasets(complete_data=complete_data,
+                       forecast_period=forecast_period,
+                       holdout_period=holdout_period,
+                       response_variable=response_variable))
+
+    observation_length = complete_data.shape[0]
+    total_variable_count = len(topic_variables)
+    predictions = np.zeros(shape=(observation_length, bootstrapIteration))
+    cv_min = np.zeros(bootstrapIteration)
+    # This is the of the implementation in R due to the specification of Python
```

Comment:
??

Reply:
???

Reply:
I guess @marcogarieri 's question was about the comment, but at least elaborate it....

My comment was poorly phrased and hastily written. Basically, the parameterisation of the exponential distribution is different in R as opposed to Python. By default, R uses `rate` while Python uses `scale`. Thus the reciprocal value of R is taken for Python so that the two implementation match.

Reply:
yeah, we were both perplexed about the comment.
Makes total sense, maybe copy what you just wrote here in the comment?